lib/src/range/range.dart

```diff
@@ -235,9 +241,15 @@ class Range<T extends Comparable<T>> {
       if (!other.lowerBounded) {
         return false;
       } else {
-        final lowerCompare = Comparable.compare(lower, other.lower);
-        if (lowerCompare > 0 || (lowerCompare == 0 && !lowerClosed &&
-            other.lowerClosed)) {
+        if (lower == null) {
+          return false;
+        }
+        if (other.lower == null) {
```

Comment:
@hovadur could you please combine the two checks above into one:

```dart
if (lower == null || other.lower == null) {
  return false;
}
```

Reply:
done